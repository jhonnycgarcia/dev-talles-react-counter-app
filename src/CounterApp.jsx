import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {

    const [counter, setCount] = useState(value);

    const increment = () => {
        // setCount(counter + 1)
        setCount((prevCount) => prevCount + 1)
    }

    const decrement = () => {
        setCount((prevCount) => prevCount - 1)
    }

    const reset = () => {
        setCount(value)
    }

    return (
        <>
            <h1>CounterApp</h1>
            <h2> {counter} </h2>
            <button onClick={increment}>+1</button>
            <button onClick={decrement}>-1</button>
            <button onClick={reset}>Reset</button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
};

CounterApp.defaultProps = {
    value: 0,
}
