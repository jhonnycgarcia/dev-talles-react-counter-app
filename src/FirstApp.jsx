
import PropTypes from 'prop-types';

const getResult = () => {
  return 'Hola'
}

export const FirstApp = ({ name, title, subtitle }) => {

  return (
    <>
      <h1>{getResult() + ' ' + title}</h1>
      <div>{subtitle}</div>
      <div>{name}</div>
    </>
  )
}

FirstApp.propTypes = {
  name: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
  name: 'Pedro',
  subtitle: 'Este es el subtítulo por defecto',
  title: 'Título por defecto',
}