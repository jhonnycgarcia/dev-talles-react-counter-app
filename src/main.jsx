import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'
// import { App } from './App'
import { FirstApp } from './FirstApp'

import './style.css'

createRoot(document.getElementById('root')).render(
    <StrictMode>
        {/* <App /> */}
        <FirstApp title='Pedro' subtitle='El primer app'/>
    </StrictMode>,
)