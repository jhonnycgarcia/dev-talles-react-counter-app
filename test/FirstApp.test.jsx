import { render } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";


describe('Pruebas en FirstApp.jsx', () => {

    // test('debe hacer match con el snapshot', () => {
    //     const title = 'Hola Mundo';
    //     const { container } = render(<FirstApp title={ title } />);
    //     expect( container ).toMatchSnapshot();
    // });

    test('debe de mostrar el titulo en un h1', () => {
        const title = 'Hola Hola Mundo';
        const { container, getByText, getByTestId } = render(<FirstApp title={ title } />);
        expect( getByText( title ) ).toBeTruthy();

        // const h1 = container.querySelector('h1');
        // expect( h1 ).toBeTruthy();
        // // expect( h1.textContent ).toBe( title );
        // expect( h1.innerHTML ).toContain( title );

        expect( getByTestId('test-title') ).toBeTruthy();
        expect( getByTestId('test-title').textContent ).toBe( title );
    });


    test('debe de mostrar el subtÃ­tulo enviado como prop', () => {
        const title = 'Hola Mundo';
        const subtitle = 'Test Subtitle';
        const { getByText } = render(<FirstApp title={ title } subtitle={ subtitle } />);
        expect( getByText( subtitle ) ).toBeTruthy();
        expect( getByText( subtitle ).textContent ).toBe( subtitle );
    });

});