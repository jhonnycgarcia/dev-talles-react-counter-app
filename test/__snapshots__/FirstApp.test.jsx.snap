// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp.jsx debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola Mundo
  </h1>
  <p>
    Este es el subt√≠tulo por defecto
  </p>
  <p>
    Pedro
  </p>
</div>
`;
